src/_utils.ts

```diff
@@ -28,15 +28,22 @@ export function isObject(value) {
 }
 
 export function matchAll(regex, string, addition) {
-  const matches = [];
-  for (const match of string.matchAll(regex)) {
-    matches.push({
+  const comments = [...string.matchAll(/\/\/.*|\/\*[\S\s]*?\*\//g)].map(
```

Comment:
Just noting that this will also match things like URLs and comments within strings:

```js
const start = "/*"
import foo from 'bar' // <- this will be ignored
const end = "*/"

// everything after start of the url will be ignored
const b = url === "https://test.com" ? await import("something") : await import("something-else")
```

Reply:
Regex is not enough to handle those cases, and that's why we made https://github.com/antfu/strip-literal for Vite and unimport. I remember we had a discussion about having it

Reply:
https://github.com/unjs/mlly/pull/56 this one. Maybe we could reuse it?

Reply:
Nice solution 👍 Even if there are some downsides (I think tokenizer sometimes fails - so we should wrap in try/catch) it should still be a good improvement.

Reply:
Thanks for reviewing! Good points both.

Suggestions:
- Maybe we can make regex more lean in them meanwhile (only comments in start of the lines to be ignored)
- Fully agree to make similar improvement of #56 to imports i think we mentioned it but was forgotten. If you want to work on it @antfu ❤️

Reply:
what a genius!

but how about this comments reg: (?<!['"`])\/\/.*|(?<!['"`])\/\*(?:(?!\/\*|\*\/)["'\s\S])*\*\/(?<!\*\/['"`])